blockchain/common_test.go

```diff
@@ -754,6 +755,34 @@ func (g *chaingenHarness) ExpectTip(tipName string) {
 	}
 }
 
+// ExpectUtxoSetState expects the provided block to be the last flushed block in
+// the utxo set state in the database.
+func (g *chaingenHarness) ExpectUtxoSetState(blockName string) {
+	g.t.Helper()
+
+	// Fetch the utxo set state from the database.
+	var gotState *utxoSetState
+	err := g.chain.db.View(func(dbTx database.Tx) error {
+		var err error
+		gotState, err = dbFetchUtxoSetState(dbTx)
+		return err
+	})
+	if err != nil {
+		g.t.Fatalf("unexpected error fetching utxo set state: %v", err)
+	}
+
+	// Validate that the state matches the expected state.
+	block := g.BlockByName(blockName)
+	wantState := &utxoSetState{
+		lastFlushHeight: block.Header.Height,
+		lastFlushHash:   block.BlockHash(),
+	}
+	if !reflect.DeepEqual(gotState, wantState) {
```

Comment:
I realize `reflect.DeepEqual` is convenient but considering its for a single call site you could drop the dependency and compare the fields directly.

Reply:
Personally, I prefer using `reflect.DeepEqual` for struct comparisons in test code as it is concise, easy to read, additionally handles the case of comparing nil pointers, and will, by default, include comparing any new struct fields as they are added.  It is also already used all over in the blockchain package test code.

I don't have an issue updating it if there is a standard or best practice we are trying to follow, though.